Dockerfile.server

```diff
@@ -84,12 +86,21 @@ COPY --chown=$user:$user packages/server ./packages/server
 
 RUN yarn install
 
+### Final image
+FROM gcr.io/distroless/nodejs:16
+
+USER nonroot
```

Comment:
Previously you used the user `node`, now it is `nonroot`. Somehow this is a bit confusing and could need an explanation.

Reply:
The user `node` is an unprivileged user that is built into the base image `nodejs:16-bullseye`, which is only used during the build stage.

The user `nonroot` is an unprivileged user that is built into the base image `nodejs:16`, which is used for the final image.

Reply:
Ok, so changing the name of the user doesn't have any ill effects, but since bind mounts have to reference an absolute path inside the container (e.g. `/home/joplin/packages/server/db-prod.sqlite` is allowed, but  `~/packages/server/db-prod.sqlite` is not), the change from `/home/joplin` to `/home/nonroot` would break existing setups that bind this file on the host. I'll overhaul my pull request to maintain compatibility with such setups.

Reply:
@tessus The compatibility issue should be resolved now. While I was at it, I removed the user `node` from the build stage since there's no benefit to running the build stage as an unprivileged user.

Reply:
Thanks for checking and the revision.